sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
I think the way forward is still to return symengine objects and try to be 100% compatible by calling into sympy for things that are not yet implemented, and convert back to symengine before returning.

This has the "1+I is a Complex class" problem. However, this can be fixed by simply converting to sympy, whenever this causes a problem in user's code. I expect there actually won't be that many places in sympy that require I to be ImaginaryUnit, as opposed to Complex. We can design a compatibility function, say, "is_imaginary_unit()", which would be implemented in the compatibility layer, and in there, if sympy is used, it would simply use `is_instance(ImaginaryUnit)`, and if symengine is used, it would query the `Complex` class. Then one just has to replace all occurrence of `is_instance(ImaginaryUnit)` in sympy with `is_imaginary_unit()`.

Now imagine some algorithm in sympy requires `1+I` to be an `Add`. Then in this very code, we simply call sympify() to convert into sympy from symengine. And we are done. Then, if we care about speed here, we can implement this whole algorithm in C++.

So I think this way we can fix all such issues. It's sort of like the Python 2 vs 3 workarounds --- Python 3 removes some classes, does some stuff differently, and so we introduce compatibility layer, and we must use it throughout. 

I think this can work. What do you think?